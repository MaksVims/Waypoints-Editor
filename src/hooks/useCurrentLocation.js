import {useEffect, useState} from "react";

export function useCurrentLocation(callback) {
  const [data, setData] = useState(null)
  const [loading, setLoading] = useState(true)

  useEffect(() => {
    navigator.geolocation.getCurrentPosition(position => {
      callback(position)
      setData(position)
      setLoading(false)
    })
  }, [])

  return [data, loading]
}