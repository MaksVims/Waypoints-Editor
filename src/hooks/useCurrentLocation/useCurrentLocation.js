import {useEffect, useState} from "react";

export function useCurrentLocation(callback) {
  const [data, setData] = useState(null)
  const [loading, setLoading] = useState(true)

  useEffect(async () => {
    if (navigator?.geolocation?.getCurrentPosition) {
      navigator?.geolocation?.getCurrentPosition(position => {
        if (typeof callback === 'function') callback(position)
        setLoading(false)
        setData(position)
      })
    }
  }, [callback])

  return [data, loading]
}