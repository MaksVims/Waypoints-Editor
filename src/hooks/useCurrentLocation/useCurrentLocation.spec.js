import {renderHook} from "@testing-library/react-hooks";
import {useCurrentLocation} from "./useCurrentLocation";


describe(">>> HOOK --- USE_CURRENT_LOCATION", function () {
  let callback

  beforeEach(() => {
    callback = jest.fn()
  })

  it('+++ should hook return correct value', async function () {
    const {result} = renderHook(() => useCurrentLocation(callback))

    expect(result.current[0]).toBe(null)
    expect(result.current[1]).toBe(true)
  });

  it('+++ should equal snapshot', function () {
    const result = renderHook(() => useCurrentLocation(callback))
    expect(result).toMatchSnapshot()
  });

})