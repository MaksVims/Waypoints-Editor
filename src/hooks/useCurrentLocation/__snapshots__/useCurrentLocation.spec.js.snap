// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> HOOK --- USE_CURRENT_LOCATION +++ should equal snapshot 1`] = `
Object {
  "rerender": [Function],
  "result": Object {
    "all": Array [
      Array [
        null,
        true,
      ],
    ],
    "current": Array [
      null,
      true,
    ],
    "error": undefined,
  },
  "unmount": [Function],
  "waitFor": [Function],
  "waitForNextUpdate": [Function],
  "waitForValueToChange": [Function],
}
`;
